--- a/src/Resources/views/storefront/layout/header/actions/cart-widget.html.twig
+++ b/src/Resources/views/storefront/layout/header/actions/cart-widget.html.twig
@@ -1,5 +1,3 @@
-{% sw_extends "@Storefront/storefront/layout/header/actions/cart-widget.html.twig" %}
-
-{% block layout_header_actions_cart_widget %}
-    ...
-{% endblock %}
+{% block layout_header_actions_cart_widget %}{% endblock %}
