--- a/src/Resources/views/storefront/layout/header/actions/cart-widget.html.twig
+++ b/src/Resources/views/storefront/layout/header/actions/cart-widget.html.twig
@@ -1,19 +1 @@
-{% sw_extends "@Storefront/storefront/layout/header/actions/cart-widget.html.twig" %}
-
-{% block layout_header_actions_cart_widget %}
-    {% if 'net' === context.taxState and 0 > page.cart.price.positionPrice %}
-        {% set cartPositionPrice = 0 %}
-    {% else %}
-        {% set cartPositionPrice = page.cart.price.positionPrice %}
-    {% endif %}
-
-    <span class="header-cart-icon">
-        {% sw_icon 'bag' %}
-    </span>
-    {% if page.cart.lineItems|length > 0 %}
-        <span class="badge bg-primary header-cart-badge">{{ page.cart.lineItems|length }}</span>
-    {% endif %}
-    <span class="header-cart-total">
-        {{ cartPositionPrice|currency }}{{ "general.star"|trans|sw_sanitize }}
-    </span>
-{% endblock %}
+{% block layout_header_actions_cart_widget %}{% endblock %}
