diff --git a/System/SalesChannel/Api/StructEncoder.php b/System/SalesChannel/Api/StructEncoder.php
index e2f2f18fe..5049699e5 100644
--- a/System/SalesChannel/Api/StructEncoder.php
+++ b/System/SalesChannel/Api/StructEncoder.php
@@ -96,7 +96,7 @@ class StructEncoder
                 continue;
             }
 
-            if (!$this->isAllowed($alias, $property, $fields) && !$fields->hasNested($alias, $property)) {
+            if (!$this->isAllowed($alias, (string)$property, $fields) && !$fields->hasNested($alias, $property)) {
                 unset($data[$property]);
 
                 continue;
@@ -129,7 +129,7 @@ class StructEncoder
                 continue;
             }
 
-            $data[$property] = $this->encodeNestedArray($struct->getApiAlias(), $property, $value, $fields);
+            $data[$property] = $this->encodeNestedArray($struct->getApiAlias(), (string)$property, $value, $fields);
         }
 
         $data['apiAlias'] = $struct->getApiAlias();
