Index: vendor/store.shopware.com/reschannableconnector/src/Core/Api/Controller/ResChannableApiController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/Core/Api/Controller/ResChannableApiController.php b/src/Core/Api/Controller/ResChannableApiController.php
--- a/src/Core/Api/Controller/ResChannableApiController.php	
+++ b/src/Core/Api/Controller/ResChannableApiController.php	(date 1691158378501)
@@ -1076,6 +1076,8 @@
             if ( method_exists($product,'getStreams') ) {
                 $streams = $product->getStreams();
 
+                $product->getMainCategories()?->filterBySalesChannelId('c31731a56382434b87b9228d484986f7')->first();
+
                 if ($streams instanceof ProductStreamCollection) {
                     $streams = $streams->getElements();
 
